common/persistence/nosql/nosqlplugin/cassandra/history_events_test.go

```diff
@@ -0,0 +1,246 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package cassandra
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin"
+	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin/cassandra/gocql"
+	"github.com/uber/cadence/common/types"
+)
+
+func TestInsertIntoHistoryTreeAndNode(t *testing.T) {
+	tests := []struct {
+		name        string
+		treeRow     *nosqlplugin.HistoryTreeRow
+		nodeRow     *nosqlplugin.HistoryNodeRow
+		setupMocks  func(session *fakeSession)
+		expectError bool
+	}{
+		{
+			name: "Successfully insert tree and node row",
+			treeRow: &nosqlplugin.HistoryTreeRow{
+				TreeID:          "treeID",
+				BranchID:        "branchID",
+				Ancestors:       []*types.HistoryBranchRange{}, // Adjusted to slice of pointers
+				CreateTimestamp: time.Now(),
+			},
+			nodeRow: &nosqlplugin.HistoryNodeRow{
+				TreeID:       "treeID",
+				BranchID:     "branchID",
+				NodeID:       1,
+				Data:         []byte("data"),
+				DataEncoding: "encoding",
+			},
+			setupMocks: func(session *fakeSession) {
+				session.mapExecuteBatchCASApplied = true
+				session.mapExecuteBatchCASErr = nil
+			},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			session := &fakeSession{
+				query: gocql.NewMockQuery(ctrl),
+			}
+			if tt.setupMocks != nil {
+				tt.setupMocks(session)
+			}
+
+			db := &cdb{session: session}
+
+			err := db.InsertIntoHistoryTreeAndNode(context.Background(), tt.treeRow, tt.nodeRow)
```

Comment:
There's a lot of if/elses in this function. Can you add more cases to cover them as well? Once the file reaches certain number of covered lines we tend to not get back to it so better to cover as much as possible as we go and call it done.

Reply:
That's my plan. I added these tests just to start with a success baseline since the file didn't exist.

Reply:
ok as long as you are planing to cover those it doesn't matter which PR. thanks